packages/core/src/Site/index.js

```diff
@@ -1103,6 +1111,8 @@ class Site {
       } else {
         isCompleted = await this.generatePagesAsyncThrottled(task.pages, progressBar);
       }
+
+      this.siteLinkManager.validateAllIntralinks();
```

Comment:
Lgtm 👍, I'll leave @ryoarmanda to the final look and merge

Just adding a note that while this preserves the previous behaviour (across rebuilds, invalid links are only logged for rebuilt files and their dependencies), perhaps we we could improve on this in a future issue / pr to persist the invalid links across rebuilds. (one more to the list for #1418)

E.g.

- `markbind s` - *all* invalid links are reported
- change some file -> only invalid links belonging to the file (or its dependencies) are reported
   - author has to scroll up to the "first batch" of messages to see the invalid links

Or
- `markbind s -o -b` - *all* invalid links are reported for the opened page (and its dependencies), then for the background pages
- change opened file -> only invalid links belonging to the file (or its dependencies) are reported
   - background pages are reported "disjointed" from the earlier batch, same issue with the author having to piece together the information again


> I think your implementation is fine too (maybe we could check first if there is any difference, functional or otherwise, on validating in real-time vs at the end?)

No preference on this for now, but doing it at the end should facilitate the 2nd item (hash checking) in #1418, where it'll likely be necessary to do it at the end (only once all hashes have been collected from all pages).